pkg/kubelet/container_bridge.go

```diff
@@ -19,16 +19,57 @@ package kubelet
 import (
 	"bytes"
 	"net"
+	"os"
 	"os/exec"
 	"regexp"
 
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
 	"github.com/golang/glog"
 )
 
 var cidrRegexp = regexp.MustCompile(`inet ([0-9a-fA-F.:]*/[0-9]*)`)
 
+func createCBR0(wantCIDR *net.IPNet) error {
+	// recreate cbr0 with wantCIDR
+	if err := exec.Command("brctl", "addbr", "cbr0").Run(); err != nil {
+		glog.Error(err)
+		return err
+	}
+	if err := exec.Command("ip", "addr", "add", wantCIDR.String(), "dev", "cbr0").Run(); err != nil {
+		glog.Error(err)
+		return err
+	}
+	if err := exec.Command("ip", "link", "set", "dev", "cbr0", "up").Run(); err != nil {
+		glog.Error(err)
+		return err
+	}
+	// restart docker
+	if util.UsingSystemdInitSystem() {
+		if err := exec.Command("systemctl", "retart", "docker").Run(); err != nil {
+			glog.Error(err)
+			// For now just log the error. The containerRuntime check will catch docker failures.
+			// TODO (dawnchen) figure out what we should do for rkt here.
```

Comment:
@fcantournet good catch. Thanks.

@roberthbailey It is still relevant. We talked about this when you guys first introducing configuring cbr0 by kubelet. One of reasons of restarting docker is killing all existing kubernetes containers which might have wrong network configurations. rkt doesn't require network containers, and rkt is daemon-less too, we need to figure out the impact here too.